v2/user_claims.go

```diff
@@ -25,7 +25,8 @@ import (
 type User struct {
 	Permissions
 	Limits
-	BearerToken bool `json:"bearer_token,omitempty"`
+	BearerToken            bool                `json:"bearer_token,omitempty"`
+	AllowedConnectionTypes map[string]struct{} `json:"allowed_connection_types,omitempty"`
```

Comment:
We would need to specify what are the possible values, but at the same time, I would not want to have a validation that enforces the values in the map to be only one of the listed values. The reason is that if the server introduces a new type (say MQTT coming), I would not want to have to update JWT first then the server.
I think that the server doing the validation is enough.

Reply:
Possible values for now could be: "STANDARD", "WEBSOCKET". We would later have "MQTT".

Reply:
Do you think we should validate the values and introduce constants?

Reply:
Using a map here looks good in go. but that's the resulting json that someone would have to produce.

 {
  "Hello": "world",
  "allowed_connection_types": {
    "foo": {}
  }
}
So I wonder if we should go with a type StringList []string in types.go.
It only adds unique strings

Reply:
Completely agree on the type, we could have a []string and the server can convert that to the map when processing the JWT. 
As for validation, see my earlier note, we could define the values in the JWT, but we need them anyway in the server when processing config files. And that's ok, server could use the JWT constant values, but it means that when we introduce a new type, we have to release a JWT version, then update the server. So I thought that we would minimize validation in JWT and do it in the server, but I could be convinced to do it in JWT.

Reply:
Maybe do constants for the values we know, just so they exist in this api, but no validation.

Reply:
StringList has accessors to remove duplicates etc.

Reply:
As for the values that can be in there, I'd have a value of LEAFNODE as well

Reply:
> Completely agree on the type, we could have a []string and the server can convert that to the map when processing the JWT.
> As for validation, see my earlier note, we could define the values in the JWT, but we need them anyway in the server when processing config files. And that's ok, server could use the JWT constant values, but it means that when we introduce a new type, we have to release a JWT version, then update the server. So I thought that we would minimize validation in JWT and do it in the server, but I could be convinced to do it in JWT.

The JWT API for reading this list could provide the map/lookup functionality to test if the connection is allowed the specified type. This saves the client from mapping the code when we already could have mapped it to do that. It will also handle the condition where the list is empty (which means all are allowed)

Reply:
We should have constants for all these, even if the list is not constrained to those constants

Reply:
I have changed to StringList and added constants, but not sure what is the best location to defined those. In types.go or directly in user_claims where I added them?